### PR TITLE
Adds DraftSubmission feature flag [pr]

### PR DESCRIPTION
**Pull request by aaronschachter**
#### What's this PR do?

This PR begins on deprecating the [`POST /campaignActivity` endpoint in Gambit Campaigns](https://github.com/DoSomething/gambit-campaigns/blob/master/documentation/endpoints/campaignActivity.md), that this app currently uses to create photo posts in Rogue. It adds a new `DraftSubmission` model that stores a user's quantity, photo URL, and why participated for a photo post topic. After all photo post properties are collected, it's sent as a text post (for now) to Rogue and the `DraftSubmission` document is deleted.

This feature is enabled when a new `DS_GAMBIT_CONVERSATIONS_DRAFT_SUBMISSION_ENABLED` config variable is set to `true`. This won't happen on production until these changes are made in a future PR:

* We send a photo post instead of a text post. Left this out to keep this PR easier to review, as it requires[ fetching the photo URL to send the file to Rogue](https://github.com/DoSomething/gambit-campaigns/blob/fd126c07f6a926591e690b1658c8bbf7ab1d4056/lib/helpers/campaignActivity.js#L25)
* Unit tests from Gambit Campaigns are ported over to add coverage to these changes

This PR also moves the `catchAll-` prefixed middleware in `lib/middleware/messages/member` into a new `topics` dir, which contains a `posts` dir to group middleware for text and photo post topics.

#### How should this be reviewed?

Locally, can set the config variable to true to confirm a DraftSubmission is created upon sending a START command while in a photo post topic -- but we don't aim to turn this on in production for a bit. So the end testing on staging can be to verify that all works as expected (posts are created via posting to the Gambit Campaigns `POST /campaignActivity` endpoint)

I'd be curious to know whether reviewers think `lilb/middleware/messages/member/topics/posts/photo/draftSubmission` should be split out into separate middleware functions, [as we do for this feature in Gambit Campaigns](https://github.com/DoSomething/gambit-campaigns/tree/master/lib/middleware/campaignActivity/photo). WIth async/await, it felt cleaner to keep in one file, although it will make unit testing a little more verbose.

#### Any background context you want to provide?

This is a big step towards [deprecating Gambit Campaigns entirely](https://github.com/DoSomething/gambit-conversations/pull/437#issuecomment-438358674), once we port over `POST /campaignActivity` into this app -- Gambit Campaigns will solely be used for getting parsed/cached topics, broadcasts, and default topic triggers aka keywords, which we aim to use GraphQL for in the not too distant future.

Once ported over, when it's time to exclude Campaign Run Id's in Rogue requests -- we can do so in a single app, which will make the switch considerably easier to coordinate (e.g. #439)

#### Relevant tickets
https://www.pivotaltracker.com/story/show/161494958

#### Checklist
- [ ] Tests added/updated for new features/bug fixes.
- [x] Tested on staging.

### CODE REVIEW

**Comment by aaronschachter** [comment]
Note: the DraftSubmissions feature currently isn't asking user to add a caption for their photo post. I've [asked about this in Slack](https://dosomething.slack.com/archives/C03V8B1HA/p1542649413131800), as it feels like an outdated requirement.

File: test/unit/lib/middleware/messages/member/topics/askYesNo.test.js [hunk 1]
```diff
@@ -9,17 +9,17 @@ const sinonChai = require('sinon-chai');
 const httpMocks = require('node-mocks-http');
 const underscore = require('underscore');
 
-const helpers = require('../../../../../../lib/helpers');
-const logger = require('../../../../../../lib/logger');
-const broadcastFactory = require('../../../../../helpers/factories/broadcast');
-const topicFactory = require('../../../../../helpers/factories/topic');
-const userFactory = require('../../../../../helpers/factories/user');
+const helpers = require('../../../../../../../lib/helpers');
```
Comment (DFurnes) [inline test/unit/lib/middleware/messages/member/topics/askYesNo.test.js]: Man, relative imports are just the worst. 😅

File: lib/middleware/messages/member/topics/posts/photo/create.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+const logger = require('../../../../../../logger');
+
+module.exports = function createPhotoPost() {
+  return async (req, res, next) => {
+    try {
+      if (!helpers.topic.isPhotoPostConfig(req.topic)) {
+        return next();
+      }
+
+      // TODO: Fetch file from url property to submit a photo post.
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply lib/middleware/messages/member/topics/posts/photo/create.js]: This was done in #444 !

File: app/models/DraftSubmission.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+/**
+ * Schema.
+ */
+const draftSubmissionSchema = new mongoose.Schema({
+  conversationId: {
+    type: mongoose.Schema.Types.ObjectId,
+    ref: 'Conversation',
+  },
+  topicId: String,
```
[the remark this answers is not in the document]
Comment (aaronschachter) [reply app/models/DraftSubmission.js]: Nope, it's a string -- we save Contentful entry id's here.

**Pull request merged**